python-sdk/pachyderm_sdk/api/worker/extension.py

```diff
@@ -0,0 +1,54 @@
+from contextlib import contextmanager
+from typing import ContextManager, Optional
+
+from dotenv import load_dotenv
+
+from ...constants import DOTENV_PATH_WORKER
+from . import WorkerStub as _GeneratedWorkerStub
+
+
+class WorkerStub(_GeneratedWorkerStub):
+
+    __error: Optional[str] = None   # used by batch_datums
+
+    @contextmanager
+    def batch_datum(self) -> ContextManager:
+        """A ContextManager that, when entered, calls the NextDatum
+        endpoint within the worker to step forward during datum batching.
+        This context manager will also prepare the environment for the user
+        and report errors that occur back to the worker.
+
+        This context manager expects to be called within an infinite while
+        loop -- see the examples section. This context can only be entered
+        from within a Pachyderm worker and the worker will terminate your
+        code when all datums have been processed.
+
+        Note: The API stub must have an open gRPC channel with the worker
+        for NextDatum to function correctly. The ``Client`` object
+        should automatically do this for the user.
+
+        Examples
```

Comment:
it slipped my mind on the datum batching example you wrote - it could be good to add an example in object detection for batch_datum() as well as the decorator

Reply:
Only having an example for `batch_all_datums` was purposeful as we want to guide users towards that workflow. `batch_datums` is provided just in case users need the added flexibility.